src/goodbase/services/evg_service.py

```diff
@@ -63,15 +64,42 @@ def analyze_build(build: Build) -> Optional[BuildStatus]:
             all_tasks=all_tasks,
         )
 
-    def check_version(self, evg_version: Version, build_checks: List[BuildChecks]) -> bool:
+    def _get_successful_tasks(self, tasks: List[Task], allow_known_failures: bool) -> set[str]:
+        return {
+            task.display_name
+            for task in tasks
+            if task.is_success() or (allow_known_failures and self._is_task_a_known_failure(task))
```

Comment:
I would make _really_ sure to read the Python language documentation for boolean expressions to make sure that it doesn't unconditionally call `_is_task_a_known_failure()` if allow_known_failures is false or the task succeeded 😉

Reply:
hmmm.....

In most languages I know they sequentially check... If the first is true, they don't continue - but I'll check.

seemingly, in python it's the same:
print("A ") if True and print("B ") else print("C ")
B
C

print("A ") if False and print("B ") else print("C ")
C

(B was not printed)